File: EbaNews.Web/Areas/Admin/Controllers/UsersController.cs
using EbaNews.Core;
using EbaNews.Services.Membership;
using EbaNews.Web.Areas.Admin.Models.Users;
using Microsoft.AspNet.Identity;
using Microsoft.AspNet.Identity.Owin;
using System;
using System.Linq;
using System.Threading.Tasks;
using System.Web;
using System.Web.Mvc;

namespace EbaNews.Web.Areas.Admin.Controllers
{
    [Authorize(Roles = Roles.Owner)]
    public class UsersController : BaseController
    {
        private readonly ApplicationUserManager userManager;

        public UsersController()
        {
            userManager = System.Web.HttpContext.Current.GetOwinContext().GetUserManager<ApplicationUserManager>();
        }

        [HttpGet]
        public ActionResult Index()
        {
            return View();
        }

        [HttpGet]
        public ActionResult GetUsers()
        {
            try
            {
                var users = userManager
                    .Users
                    .ToList()
                    .Select(user => new UserViewModel
                    {
                        Id = user.Id,
                        Name = user.UserName,
                        Email = user.Email,
                        Roles = userManager.GetRoles(user.Id)
                    });

                return Json(users, JsonRequestBehavior.AllowGet);
            }
            catch (Exception ex)
            {
                return InternalError(ex.Message);
            }
        }

        [HttpPost]
        public async Task<ActionResult> SwitchRole(SwitchRoleViewModel model)
        {
            try
            {
                if (await userManager.IsInRoleAsync(model.UserId, model.Role))
                {
                    await userManager.RemoveFromRoleAsync(model.UserId, model.Role);
                }
                else
                {
                    await userManager.AddToRoleAsync(model.UserId, model.Role);
                }

                return Json(userManager.GetRoles(model.UserId));
            }
            catch (Exception ex)
            {
                return InternalError(ex.Message);
            }
        }
    }
}

File: EbaNews.Web/Scripts/Areas/Admin/usersController.js
var ViewModel = function () {
    var self = this;
    self.users = ko.observableArray();
    self.loading = ko.observable(false);

    self.user = function (id, name, email, roles) {
        this.Id = id;
        this.Name = name;
        this.Email = email;
        this.Roles = ko.observableArray(roles);
    };

    self.map = function (data) {
        for (var i = 0; i < data.length; i++) {
            var user = new self.user(data[i].Id, data[i].Name, data[i].Email, data[i].Roles);
            self.users.push(user);
        }
    };

    self.startLoading = function () {
        self.loading(true);
    };

    self.stopLoading = function () {
        self.loading(false);
    };

    self.isInRole = function (user, role) {
        return user.Roles.indexOf(role) !== -1;
    };

    self.getUsers = function () {
        self.startLoading();
        $.get("/mngmnt/users/get", function (response) {
            self.map(response);
            self.stopLoading();
        });
    };

    self.switchAdminRole = function (user) {
        self.switchRole(user, "admin");
    };

    self.switchOwnerRole = function (user) {
        self.switchRole(user, "owner");
    };

    self.switchRole = function (user, role) {
        var data = {
            UserId: user.Id,
            Role: role
        };

        $.post("/mngmnt/users/role/switch", data, function (response) {
            user.Roles(response);
        });
    };

    self.getUsers();
};

ko.applyBindings(new ViewModel());